doobie/src/main/scala/za/co/absa/fadb/doobie/DoobieFunction.scala

```diff
@@ -39,25 +57,71 @@ trait DoobieFunctionBase[R] {
  *  @tparam I the input type of the function
  *  @tparam R the result type of the function
  */
-private[doobie] trait DoobieFunction[I, R] extends DoobieFunctionBase[R] {
+trait DoobieFunction[I, R, F[_]] extends DoobieFunctionBase[R] {
+
   /**
-   *  Generates a Doobie `Fragment` representing the SQL query for the function.
-   *
+   *  Function that generates a sequence of `Fragment`s representing the SQL query from input values for the function.
+   *  @return the sequence of `Fragment`s representing the SQL query
+   */
+  def toFragmentsSeq: I => Seq[Fragment]
```

Comment:
perhaps consider this 

```suggestion
  def valuesToFragmentsSeq: I => Seq[Fragment]
```

Reply:
oh no, I understood it now - ignore this please